src/celeritas/neutron/xs/NeutronInelasticMicroXsCalculator.hh

```diff
@@ -0,0 +1,58 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/xs/NeutronInelasticMicroXsCalculator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "celeritas/neutron/data/NeutronInelasticData.hh"
+
+#include "NeutronMicroXsCalculator.hh"
+
+namespace celeritas
+{
+class NeutronInelasticMicroXsCalculator;
+
+//---------------------------------------------------------------------------//
+/*!
+ * XsData_traits for NeutronInelasticRef.
+ */
+template<>
+struct XsData_traits<NeutronInelasticMicroXsCalculator>
+{
+    using ParamsRef = NeutronInelasticRef;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate neutron inelastic cross sections from NeutronInelasticData.
+ */
+class NeutronInelasticMicroXsCalculator
+    : public NeutronMicroXsCalculator<NeutronInelasticMicroXsCalculator>
+{
```

Comment:
I don't understand this class design with the combination of inheritance, friends, traits, and CRTP. Wouldn't it be simpler to just template `NeutronMicroXsCalculator` on `ParamsRef`? Or since the structures of all the `ParamsRef` are essentially identical (shared_.micro_xs, shared_.reals), create a new class `NeutronMicroXsData<W,M>` that is used by both inelastic and elastic? Then you wouldn't have to template at all.

Reply:
@sethrj You are right that `friend` is not necessary for this case (as the derives do not need to access any member of the mother in this unusual case).  The motivation was to not to use a templated (macro) of tempated (micro) parameter pass, but still to share a common method of the microscopic xs calculation (top on the already templated macroscopic  xs calculator) for different neutron processes.  I will think about the alternative and try to come up with a different solution if you do not like this.

Reply:
Is there ever going to be a difference between the calculation method for the different neutron micro XS? So far they're all "generic grid" interpolation... I think it would be much more sensible to reuse a single class and data type... maybe we can discuss in person tomorrow?